java-extras/src/main/java/org/triplea/yaml/YamlUtils.java

```diff
@@ -0,0 +1,60 @@
+package org.triplea.yaml;
+
+import com.google.common.base.Strings;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import lombok.experimental.UtilityClass;
+import org.snakeyaml.engine.v2.api.Dump;
+import org.snakeyaml.engine.v2.api.DumpSettings;
+import org.snakeyaml.engine.v2.api.Load;
+import org.snakeyaml.engine.v2.api.LoadSettings;
+import org.snakeyaml.engine.v2.exceptions.YamlEngineException;
+
+/**
+ * Methods useful for working reading and writing YAML data. This class largely is a wrapper around
+ * snake yaml.
+ */
+@UtilityClass
+public class YamlUtils {
+  public <T> T readYaml(final String inputString) {
+    final var inputStream =
+        new ByteArrayInputStream(Strings.nullToEmpty(inputString).getBytes(StandardCharsets.UTF_8));
+    return readYaml(inputStream);
```

Comment:
I must admit that this is an elegant way to re-use the code, but I wonder if we could've achieved the same thing by passing either `load.loadFromString` or `load.loadFromInputStream` as a lambda to a common helper function

Reply:
Can you elaborate, I don't quite follow how the common helper function would look.